### DISCUSSION TITLE
Any update on Logpoints?

### DISCUSSION THREAD

**Discussion by TryerGit** [microsoft/vscode-cpptools#8163] [Q&A] [answered]
This is with possible reference to this issue: https://github.com/microsoft/vscode-cpptools/issues/2591

I am able to confirm that out of the box on linux using gdb does indeed provide log output in the Debug Console window. I was wondering if there was any plan to provide for this functionality on Windows using the Visual Studio compiler toolchain on Windows using CL.exe and MSBuild.exe? I am able to confirm that logpoints do not work on the latter (Windows/VC++ compiler/MsBuild.exe).

Additionally, is there any command/keyboard shortcut to scroll up and down the debug console? At present, out of the box, only `ctrl-shift-y` seems to have a default binding that toggles focus on the debug console. There does not seem to be any command that scrolls up and down the debug console. Just wanted to check and confirm that the only way to scroll up and down this debug console is the mouse.

Thank you.

**Comment by sean-mcmanus** [comment]
@WardenGnaw Do you know?

**Comment by WardenGnaw** [comment, marked as answer]
>  I was wondering if there was any plan to provide for this functionality on Windows using the Visual Studio compiler toolchain on Windows using CL.exe and MSBuild.exe? I am able to confirm that logpoints do not work on the latter (Windows/VC++ compiler/MsBuild.exe).

This is currently in the backlog. `cppvsdbg` is a different debug adapter and does not have this implemented with the Debug Adapter Protocol. 

> Additionally, is there any command/keyboard shortcut to scroll up and down the debug console? At present, out of the box, only ctrl-shift-y seems to have a default binding that toggles focus on the debug console. 

The debug console is owned the VS Code, and you will want to ask them about this question.

**Reply by TryerGit** [reply]
Thanks. One way to reduce / minimize the need for scrolling the debug console is to prefix the logoutput with some persistent string and then filter on the string in the top right. Otherwise, gdb is very verbose about the breakpoints/other output that it shows that the user's log output is long lost. I will try to follow up with the vscode folks.
